src/if_statements.py

```diff
@@ -570,6 +603,216 @@ def try_make_if_condition(
     return (cond, if_node, else_node)
 
 
+def try_build_irregular_switch(
+    context: Context, start: Node, end: Node
+) -> Optional[SwitchStatement]:
+    """
+    Look for irregular switch-like structures from nested ConditionalNodes & SwitchNodes.
+    If one is found, return the corresponding SwitchStatement; otherwise, return None.
+
+    Both IDO & GCC can convert switch statements into a tree of comparisons & jump tables.
+    We try to identify the largest contiguous tree of ConditionalNodes & SwitchNodes that
+    all compare the same variable (`var_expr`). SwitchNodes and `(x == N)` ConditionalNodes
+    represent jumps to specific case labels, whereas comparisons like `(x < N)` are
+    primarily used to manage the overall tree depth.
+    """
+    # The start node must either be an if or a switch
```

Comment:
```suggestion
    # The start node must either be an if
```
and you can also remove the `start_block_info.switch_control is not None` case below